cmd/api/src/daemons/datapipe/datapipe.go

```diff
@@ -104,7 +102,17 @@ func (s *Daemon) setAnalysisRequested(requested bool) {
 	s.analysisRequested = requested
 }
 
+func (s *Daemon) finishAnalysis() {
+	// Ensure that the user-requested analysis switch is flipped back to false
+	s.setAnalysisRequested(false)
```

Comment:
My understanding is that analysis can be requested again while already in progress (like when a file ingest or collection completes while analysis is currently underway.)

Does this changeset account for that scenario?

Reply:
In my brain you'd need to turn this switch off as soon as analysis _begins_ processing, and allow it to be flipped back on midway through.

Reply:
We talked about this one before and the previous location was at the end of the function which was also incorrect and meant that we didn't toggle to false if analysis was set to disabled. Of course, we ended up simplifying even more for FOSS since there's no real reason anyone would want to disable analysis here.

I think your proposal makes sense though, and perhaps we should be clearing it at the start to ensure that any analysis requests that get queued during an analysis run aren't wiped. The goal here was to make the logic more visible, and I think we found a few areas that could potentially be better as a result. I'll defer to @zinic on what the correct course of action is for this PR, but I think it will at least be worth the discussion in a separate ticket.

Reply:
I hadn't considered the case where the user flips the latch during an analysis run. I'll update this accordingly.

Reply:
This has been patched as per Eli's suggestion